Course/Course/Presentation/Dates/CourseDatesViewModel.swift

```diff
@@ -91,4 +94,49 @@ public class CourseDatesViewModel: ObservableObject {
             errorMessage = CourseLocalization.Error.componentNotFount
         }
     }
+    
+    @MainActor
+    func shiftDueDates(courseID: String, withProgress: Bool = true) async {
+        isShowProgress = withProgress
+        dueDatesShifted = false
+        do {
+            try await interactor.shiftDueDates(courseID: courseID)
+            NotificationCenter.default.post(name: .shiftCourseDates, object: courseID)
+            isShowProgress = false
+        } catch let error {
+            isShowProgress = false
+            if error.isInternetError || error is NoCachedDataError {
+                errorMessage = CoreLocalization.Error.slowOrNoInternetConnection
+            } else {
+                errorMessage = CoreLocalization.Error.unknownError
+            }
+        }
+    }
+    
+    deinit {
+        let center = NotificationCenter.default
+        guard let shiftCourseDatesObserver = self.shiftCourseDatesObserver else { return }
+        center.removeObserver(shiftCourseDatesObserver)
+    }
+}
+
+extension CourseDatesViewModel {
+    func addNotificationObserver() {
+        let center = NotificationCenter.default
+        shiftCourseDatesObserver = center.addObserver(forName: .shiftCourseDates,
+                                               object: nil,
+                                               queue: nil) { [weak self] notification in
+            if let courseID = notification.object as? String {
+                guard let strongSelf = self else {
+                    return
+                }
+                Task {
```

Comment:
What if we can define `addNotificationObserver` as MainActor so we don't have to define tasks here and even we can omit `DispatchQueue`

Reply:
If we designate it with `@MainActor` and eliminate the `Task` closure, we would have to utilize an `async` function here. However, this change leads to certain exceptions. To resolve this issue, we may consider employing `addObserver(forName:object:queue:using:)`. Currently, this method is not being utilized for notifications in the application.